source/extensions/transport_sockets/tls/cert_validator/default_validator.cc

```diff
@@ -206,13 +201,12 @@ int DefaultCertValidator::doVerifyCertChain(
     }
   }
 
-  Envoy::Ssl::ClientValidationStatus validated = verifyCertificate(
-      &leaf_cert,
-      transport_socket_options &&
-              !transport_socket_options->verifySubjectAltNameListOverride().empty()
-          ? transport_socket_options->verifySubjectAltNameListOverride()
-          : verify_subject_alt_name_list_,
-      subject_alt_name_matchers_);
+  Envoy::Ssl::ClientValidationStatus validated =
+      verifyCertificate(&leaf_cert,
+                        transport_socket_options != nullptr
```

Comment:
Just one thing to discuss with reviewers here:
This function arg becomes optional after my change.
Other than current code, there are also some other options 1) absl::optional 2) pointer and using nullptr to indicate "not exist".
I currently use an empty vector approach because 1)it is a lightweight approach from implementation perspective 2) there is a !verify_san_list.empty() check inside verifyCertificate(), which basically provides the same functionality to absl::optional's has_value. And this is better because absl::optional doesn't support pass by const ref and will cause a copy.

If this path is performance critical, using pointer/nullptr might be a slightly preferred approach. Please let me know what do you think. Thanks!